client/app/models/calendar.coffee

```diff
@@ -0,0 +1,28 @@
+Tag = require './tag'
+
+module.exports = class Calendar extends Tag
+
+    initialize: (options) ->
+        super options
+
+        # When created via Data System, a calendar should not have any
+        # color assigned. So we set it at initialization and will try
+        # to save it as soon as a change is made on the calendar.
+        color = @get 'color'
+        @hasColorDefined =
+            color and color.match /^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/
+
+        if not @hasColorDefined
+            # Temporary color until a save is made
+            @set 'color', ColorHash.getColor @get 'name'
+
+
+    # Take proffit of a saving to save the color.
+    save: (attributes, options) ->
+        attributes = attributes ?= {}
+        if not @hasColorDefined and not attributes?.color?
+            attributes.color = @get 'color'
```

Comment:
I don't set it to `true` when the color is set in the client because its role is to indicate that the client-defined color has to be sync with the DB, so I "force" the color attibute in any `save` call.

As I am explaining, I realise that I should set `hasColorDefined` to `true` in a success callback.